rerun_cpp/src/rerun/as_components.hpp

```diff
@@ -0,0 +1,80 @@
+#pragma once
+
+#include "component_batch.hpp"
+#include "indicator_component.hpp"
+
+namespace rerun {
+    /// The AsComponents trait is used to convert a type into a list of serialized component.
+    ///
+    /// It is implemented for various built-in types as well as collections of components.
+    /// You can build your own archetypes by implementing this trait.
+    /// Anything that implements `AsComponents` can be logged to a recording stream.
+    template <typename T>
+    struct AsComponents {
+        template <typename T2>
+        struct NoAsComponentsFor : std::false_type {};
+
+        // TODO(andreas): This should also mention an example of how to implement this.
+        static_assert(
+            NoAsComponentsFor<T>::value,
```

Comment:
This reads backwards to me: "assert that there is no `AsComponents` for T"

I think `NoAsComponentsFor` should be called `HasAsComponents` or something?